### DISCUSSION TITLE
FieldArray how to handle registration

### DISCUSSION THREAD

**Discussion by cargallo** [react-hook-form/react-hook-form#4446] [Q&A]
Hi, I have this form. It was complex so I decided to split a part of it into another dialog box with another react hook form to validate it. 

This is the main form

![image](https://user-images.githubusercontent.com/13320290/111396254-740e1700-869d-11eb-8d38-29916c969235.png)

This is the dialog form

![image](https://user-images.githubusercontent.com/13320290/111396352-a6b80f80-869d-11eb-983f-2418ae1b4198.png)

So when I get back to the original page I'm doing an append, or remove/insert to update, or remove when click on trash. 

I do not have any fieldsarray to input in the main form, they are just added with append, remove, insert.

I searched for solutions and readed the documentation so what I finally did is this, I need to know if this is ok. It's working but this seems really strange.

![image](https://user-images.githubusercontent.com/13320290/111396603-3362cd80-869e-11eb-84b5-486274bcd814.png)

Thanks in advance!

**Comment by bluebill1049** [comment]
They look legit for hidden input, in v7 this will no longer need to happen, we will store data in the hook form.

**Reply by cargallo** [reply]
The code inside the red square, I have to wrap input with Controller because with standard input it didn't work, do not know why. But the difference is that medioContacto is an array. Thanks for response!
